app/src/processing/app/contrib/StatusPanel.java

```diff
@@ -21,34 +21,197 @@
 */
 package processing.app.contrib;
 
+import java.awt.BorderLayout;
 import java.awt.Color;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.text.DateFormat;
+import java.util.Date;
+
+import javax.swing.GroupLayout;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
 import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JProgressBar;
+import javax.swing.JTextPane;
+import javax.swing.LayoutStyle;
+import javax.swing.SwingConstants;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.HyperlinkEvent;
+import javax.swing.event.HyperlinkListener;
+
+import processing.app.Base;
+import processing.app.Language;
+import processing.app.Toolkit;
 
 
-class StatusPanel extends JLabel {
+
+class StatusPanel extends JPanel {
+  
+  final int BUTTON_WIDTH = 20;
   
-  public StatusPanel() {
-    super(" ");  // need to have some size
-//    setBackground(null);
-//    setBorder(null);
+  JTextPane label;
+  JButton installButton;
+  JProgressBar installProgressBar;
+  JLabel updateLabel;
+  JButton updateButton;
+  JButton removeButton;
+  
+  ContributionListing contributionListing = ContributionListing.getInstance();
+  ContributionManagerDialog contributionManagerDialog;
+  
+  public StatusPanel(int width, ContributionManagerDialog contributionManagerDialog) {
+    final int BUTTON_WIDTH = 150;
+    this.contributionManagerDialog  = contributionManagerDialog;
+    label = new JTextPane();
+    label.setEditable(false);
+    label.setOpaque(false);
+    label.setContentType("text/html");
+    label.addHyperlinkListener(new HyperlinkListener() {
+
+      @Override
+      public void hyperlinkUpdate(HyperlinkEvent e) {
+        if (e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
+          if (e.getURL() != null) {
+            Base.openURL(e.getURL().toString());
+          }
+        }
+      }
+    });
+    installButton = new JButton("Install");
+    installButton.addActionListener(new ActionListener() {
+      
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        StatusPanel.this.contributionManagerDialog.getActiveTab().contributionListPanel
+          .getSelectedPanel().install();
+      }
+    });
+    installProgressBar = new JProgressBar();
+    updateLabel = new JLabel("  ");
+    updateButton = new JButton("Update");
+    updateButton.addActionListener(new ActionListener() {
+      
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        StatusPanel.this.contributionManagerDialog.getActiveTab().contributionListPanel.getSelectedPanel().update();
```

Comment:
Yes, keep things to 80 chars. If it needs two lines, you can do an assignment on one line and then use that variable on the next, rather than the huge chaining stuff. More importantly, if the chain is that long, there's something wrong in the logic, or there needs to be a more sensible set of accessors.